src/server/input/input_probe.cpp

```diff
@@ -49,6 +49,44 @@ mir::UniqueModulePtr<mi::Platform> create_input_platform(
 
     return result;
 }
+
+
+auto select_platforms_from_list(
```

Comment:
Could we refactor this together with the `default_configuration.cpp` part to avoid the copy-paste?

Reply:
Most of the body here is the same?

https://github.com/MirServer/mir/blob/7f057531eed603203b925cf9af201c58ff46c163/src/server/graphics/default_configuration.cpp#L81-L131

Reply:
@Saviq There is some _slightly_ different logic when it comes to graphics. On the input side, we always only select one platform, while for graphics we select multiple potentially

Reply:
We could share it, but I feel it would just complicate things more than it needs to